mllib/src/main/scala/org/apache/spark/ml/classification/ProbabilisticClassifier.scala

```diff
@@ -200,22 +200,17 @@ abstract class ProbabilisticClassificationModel[
     if (!isDefined(thresholds)) {
       probability.argmax
     } else {
-      val thresholds: Array[Double] = getThresholds
-      val probabilities = probability.toArray
+      val thresholds = getThresholds
       var argMax = 0
       var max = Double.NegativeInfinity
       var i = 0
       val probabilitySize = probability.size
       while (i < probabilitySize) {
-        if (thresholds(i) == 0.0) {
-          max = Double.PositiveInfinity
+        // thresholds are all > 0, excepting that at most one may be 0
+        val scaled = probability(i) / thresholds(i)
+        if (scaled > max) {
```

Comment:
Ok, I think that's reasonable behavior. Is it better to handle the zero threshold case in the code explicitly? It confused me at first, and I had to refer to divide by zero behavior in scala to understand the code.